crates/biome_css_parser/src/syntax/mod.rs

```diff
@@ -79,13 +80,192 @@ pub(crate) fn parse_rule_block(p: &mut CssParser) -> ParsedSyntax {
     }
     let m = p.start();
     p.expect(T!['{']);
-    let list = p.start();
-    list.complete(p, CSS_DECLARATION_LIST);
+    parse_declaration_list(p);
     p.expect(T!['}']);
 
     Present(m.complete(p, CSS_BLOCK))
 }
 
+pub(crate) fn parse_declaration_list(p: &mut CssParser) {
+    let m = p.start();
+
+    // first parse declaration item
+    parse_declaration_item(p);
+    loop {
+        if !p.eat(T![;]) {
+            break;
+        }
+        parse_declaration_item(p);
+    }
+    m.complete(p, CSS_DECLARATION_LIST);
+}
+
+// CssDeclaration =
+// 	name: CssIdentifier | CssCustomProperty
+// 	':'
+// 	value: AnyCssValue
+// 	important: CssDeclarationImportant?
+#[inline]
+pub(crate) fn parse_declaration_item(p: &mut CssParser) {
+    if !is_at_identifier(p) {
+        return;
+    }
+    let m = p.start();
+    // name
+    parse_regular_identifier(p).ok();
+
+    p.expect(T![:]);
+    loop {
```

Comment:
Why do we need loop here?
Is it possible to ha a list here?

Reply:
I think this should be done with ParseSeparatedList just like the one above, don't you think?

Reply:
Do we need parse a list here? 
Our gramma doesn't have a list as value now:

```
CssDeclaration =
	name: CssDeclarationName
	':'
	value: AnyCssValue
	important: CssDeclarationImportant?
```

Reply:
I think this grammar is wrong
I will fix them tomorrow
The following is the correct grammar
`flex: 1 1 0`
` background: url(something) 100%`
It is a compent value list
https://drafts.csswg.org/css-syntax/#consume-declaration